src/non_nil.rs

```diff
@@ -0,0 +1,84 @@
+//! A wrapper type for nil UUIDs that provides a more memory-efficient
+//! `Option<NonNilUuid>` representation.
+
+use std::{fmt, num::NonZeroU128};
+
+use crate::Uuid;
+
+/// A UUID that is guaranteed not to be the nil UUID.
+///
+/// This is useful for representing optional UUIDs more efficiently, as `Option<NonNilUuid>`
+/// takes up the same space as `Uuid`.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct NonNilUuid(NonZeroU128);
```

Comment:
An associated `NonNilUuid::get()` that returned a `Uuid` would be helpful here I think.

Sorry, I think I might have lead you astray with an earlier comment about not having any methods instead of conversions. What I meant there was not having any methods like `new_v4`, `get_timestamp`, `get_version` etc.

Reply:
fixed, and thanks for your valuable comments.